server_utils/server_struct.go

```diff
@@ -0,0 +1,36 @@
+/***************************************************************
+ *
+ * Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+package server_utils
+
+import "github.com/pelicanplatform/pelican/director"
+
+type (
+	XRootDServer struct {
```

Comment:
Convert this to an interface and have the implementations with composition, instead of a struct with a function pointer.

That is,

```
type XRootDServer interface {
  GetNamespaceAds func() []director.NamespaceAd
  CreateAdvertisement func(string, string, string) (director.OriginAdvertise, error)
}
```